artifactory/commands/dotnet/dotnetcommand.go

```diff
@@ -191,58 +168,77 @@ func changeWorkingDir(newWorkingDir string) (string, error) {
+	content := dotnet.ConfigFileTemplate
+	_, err := configFile.WriteString(content)
 	if err != nil {
-		return err
+		return errorutils.CheckError(err)
 	}
-	// To prevent NuGet prompting for credentials
-	err = os.Setenv("NUGET_EXE_NO_PROMPT", "true")
+	// We need to close the config file to let the toolchain modify it.
+	err = configFile.Close()
 	if err != nil {
 		return errorutils.CheckError(err)
 	}
+	return addSourceToNugetConfig(cmdType, configFile.Name(), sourceUrl, user, password)
+}
 
-	err = dc.prepareConfigFile(cmd, configDirPath)
-	if err != nil {
-		return err
-	}
-	err = io.RunCmd(cmd)
+// Runs nuget sources add command
+func addSourceToNugetConfig(cmdType dotnet.ToolchainType, configFileName, sourceUrl, user, password string) error {
+	cmd, err := dotnet.CreateDotnetAddSourceCmd(cmdType, sourceUrl)
 	if err != nil {
 		return err
 	}
 
-	return nil
+	flagPrefix := cmdType.GetTypeFlagPrefix()
+	cmd.CommandFlags = append(cmd.CommandFlags, flagPrefix+"configfile", configFileName)
+	cmd.CommandFlags = append(cmd.CommandFlags, flagPrefix+"name", SourceName)
+	cmd.CommandFlags = append(cmd.CommandFlags, flagPrefix+"username", user)
+	cmd.CommandFlags = append(cmd.CommandFlags, flagPrefix+"password", password)
+	output, err := io.RunCmdOutput(cmd)
+	log.Debug("Running command: Add sources. Output:", output)
+	return err
 }
 
 // Checks if the user provided input such as -configfile flag or -Source flag.
 // If those flags were provided, NuGet will use the provided configs (default config file or the one with -configfile)
 // If neither provided, we are initializing our own config.
-func (dc *DotnetCommand) prepareConfigFile(cmd *dotnet.Cmd, configDirPath string) error {
-	cmdFlag := cmd.GetToolchain().GetTypeFlagPrefix() + "configfile"
+func (dc *DotnetCommand) prepareConfigFile(cmd *dotnet.Cmd) (func() error, error) {
+	// Use temp dir to save config file, so that config will be removed at the end.
+	tempDirPath, err := fileutils.CreateTempDir()
+	if err != nil {
+		return nil, err
+	}
+
+	dc.solutionPath, err = changeWorkingDir(dc.solutionPath)
+	if err != nil {
+		return nil, err
+	}
+	cmdFlag := dc.GetToolchain().GetTypeFlagPrefix() + "configfile"
 	currentConfigPath, err := getFlagValueIfExists(cmdFlag, cmd)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	if currentConfigPath != "" {
-		return nil
+		return nil, nil
 	}
 
 	cmdFlag = cmd.GetToolchain().GetTypeFlagPrefix() + "source"
 	sourceCommandValue, err := getFlagValueIfExists(cmdFlag, cmd)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	if sourceCommandValue != "" {
-		return nil
+		return nil, nil
 	}
 
-	configFile, err := dc.InitNewConfig(configDirPath)
+	configFile, err := dc.InitNewConfig(tempDirPath)
 	if err == nil {
 		cmd.CommandFlags = append(cmd.CommandFlags, cmd.GetToolchain().GetTypeFlagPrefix()+"configfile", configFile.Name())
 	}
-	return err
+	return func() error {
+		return fileutils.RemoveTempDir(tempDirPath)
```

Comment:
Notice that this cleanup function is returned only if the function is returned from here. This means that the cleanup will not happen in some cases (see the previous return statements). One way to resolve this is to use named returned values, and have this function created and set to the returned value right after the temp dir is created.